core/src/main/java/com/google/zxing/datamatrix/decoder/DecodedBitStreamParser.java

```diff
@@ -348,6 +375,7 @@ private static void decodeTextSegment(BitSource bits, StringBuilder result) thro
             } else {
               switch (cValue) {
                 case 27: // FNC1
+                  fnc1positions.put(result.length(), null);
```

Comment:
Are the values ever anything except null? if it is used as a Set, use a Set<Integer>

Reply:
Ok. Just didn't remember Set also has a hash implementation, with hashing being important here for speed and space.